src/rand.rs

```diff
@@ -21,9 +22,100 @@ impl Distribution<Decimal> for Standard {
     }
 }
 
-#[test]
-fn has_random_decimal_instances() {
-    let mut rng = rand::rngs::OsRng;
-    let random: [Decimal; 32] = rng.gen();
-    assert!(random.windows(2).any(|slice| { slice[0] != slice[1] }));
+impl SampleUniform for Decimal {
+    type Sampler = DecimalSampler;
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct DecimalSampler {
+    mantissa_sampler: UniformInt<i128>,
+    scale: u32,
+}
+
+impl UniformSampler for DecimalSampler {
+    type X = Decimal;
+
+    #[inline]
+    fn new<B1, B2>(low: B1, high: B2) -> Self
+    where
+        B1: SampleBorrow<Self::X> + Sized,
+        B2: SampleBorrow<Self::X> + Sized,
+    {
+        let high = *high.borrow();
```

Comment:
Could you not just subtract `Decimal::ONE`? It should be faster since it'll take an optimal path.

Reply:
Actually, I was a bit naive here - you're only removing one from the raw mantissa as opposed to the integral portion. This is an interesting challenge for a decimal number since scale is a relevant factor.

Reply:
As you determined, I can not and should not subtract `Decimal::ONE`, because what I'm doing here is "subtract one from the least significant digit", as opposed to "subtract 1".

I assert that we can be confident that the value is _not_ `Decimal::MIN` because of the check that requires `low < high`. There's no way for `high` to be `Decimal::MIN` without triggering _that_ error.

Reply:
Fixed in 5bbe252.